sky/provision/aws/instance.py

```diff
@@ -184,20 +187,134 @@ def cleanup_ports(
+    cluster_name_on_cloud: str,
+    ports: List[str],
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> None:
+    """See sky/provision/__init__.py"""
+    assert provider_config is not None, (cluster_name_on_cloud, provider_config)
+    region = provider_config['region']
+    ec2 = _default_ec2_resource(region)
+    sg_name = provider_config['security_group']['GroupName']
+    filters = [
+        {
+            'Name': 'instance-state-name',
+            # exclude 'shutting-down' or 'terminated' states
+            'Values': ['pending', 'running', 'stopping', 'stopped'],
+        },
+        *_cluster_name_filter(cluster_name_on_cloud),
+    ]
+    instances = _filter_instances(ec2,
+                                  filters,
+                                  included_instances=None,
+                                  excluded_instances=None)
+    instance_list = list(instances)
+    if not instance_list:
+        logger.warning(
+            f'Instance with cluster name {cluster_name_on_cloud} not found. '
+            f'Skip creating security group.')
+        return None
+    sg = _get_sg_from_name(ec2, sg_name)
+    if sg is None:
+        logger.warning('Find new security group failed. Skip open ports.')
+        return
+    # For multinode cases, we need to open ports for all instances.
+    for instance in instance_list:
+        _maybe_move_to_new_sg(instance, sg)
+
+    existing_ports: Set[int] = set()
+    for existing_rule in sg.ip_permissions:
+        # Skip any non-tcp rules.
+        if existing_rule['IpProtocol'] != 'tcp':
+            continue
+        # Skip any rules that don't have a FromPort or ToPort.
+        if 'FromPort' not in existing_rule or 'ToPort' not in existing_rule:
+            continue
+        existing_ports.update(
+            range(existing_rule['FromPort'], existing_rule['ToPort'] + 1))
+    ports_to_open = resources_utils.port_set_to_ranges(
+        resources_utils.port_ranges_to_set(ports) - existing_ports)
+
+    ip_permissions = []
+    for port in ports_to_open:
+        if port.isdigit():
+            from_port = to_port = port
+        else:
+            from_port, to_port = port.split('-')
+        ip_permissions.append({
+            'FromPort': int(from_port),
+            'ToPort': int(to_port),
+            'IpProtocol': 'tcp',
+            'IpRanges': [{
+                'CidrIp': '0.0.0.0/0'
+            }],
+        })
+
+    # For the case when every new ports is already opened.
+    if ip_permissions:
+        sg.authorize_ingress(IpPermissions=ip_permissions)
+
+
+def cleanup_ports(
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> None:
+    """See sky/provision/__init__.py"""
+    assert provider_config is not None, cluster_name_on_cloud
+    region = provider_config['region']
+    ec2 = _default_ec2_resource(region)
+    sg_name = provider_config['security_group']['GroupName']
+    if cluster_name_on_cloud not in sg_name:
```

Comment:
This is also valid for clusters launched before this PR, since their security group name doesn't change.

Reply:
This is a bit hacky (information leakage). Can we move the default name to `clouds.AWS` and use the constant there?

Reply:
Done. thanks!